### PR TITLE
Source Stripe: add stream accounts

### LINKED ISSUE #9167: Source Stripe: add stream `accounts`
## Tell us about the problem you're trying to solve
Add new stream: https://stripe.com/docs/api/accounts
Requested by [slack convo](https://airbytehq-team.slack.com/archives/C019WJFR1JT/p1640717898117200)

## Describe the solution you’d like
This endpoint is important to a community user.

## Describe the alternative you’ve considered or used
Don't have

## Are you willing to submit a PR?
Not this time.

### PR DESCRIPTION
**Pull request by haithem-souala**
## What
Closes #9167 

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

**Comment by YowanR** [comment]
@marcosmarxm @sajarin Can you assist here, please?

File: airbyte-integrations/connectors/source-stripe/source_stripe/schemas/accounts.json [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    },
+    "business_type": {
+      "description": "The business type.",
+      "enum": ["company", "government_entity", "individual", "non_profit"],
+      "nullable": true,
+      "type": "string",
+      "x-stripeBypassValidation": true
+    },
+    "capabilities": {
+      "$ref": "#/components/schemas/account_capabilities"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-stripe/source_stripe/schemas/accounts.json:21]: Please add the correct data for capabilities.

File: airbyte-integrations/connectors/source-stripe/source_stripe/schemas/accounts.json [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      "type": "string"
+    },
+    "individual": {
+      "$ref": "#/components/schemas/person"
+    },
+    "metadata": {
+      "additionalProperties": {
+        "maxLength": 500,
+        "type": "string"
+      },
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-stripe/source_stripe/schemas/accounts.json:105]: Please clean up the schema to follow the standard.

Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-stripe/source_stripe/schemas/accounts.json:105]: Check other Stripe streams.

File: airbyte-integrations/connectors/source-stripe/Dockerfile [hunk 3]
```diff
@@ -12,5 +12,5 @@ COPY main.py ./
 ENV AIRBYTE_ENTRYPOINT "python /airbyte/integration_code/main.py"
 ENTRYPOINT ["python", "/airbyte/integration_code/main.py"]
 
-LABEL io.airbyte.version=3.0.0
+LABEL io.airbyte.version=3.1.0
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-stripe/Dockerfile:15]: You only adding a new stream which is a patch update. Correct this to `3.0.1`

File: docs/integrations/sources/stripe.md [hunk 4]
```diff
@@ -69,19 +69,21 @@ The Stripe source connector supports the following streams:
...
 - [Transfers](https://stripe.com/docs/api/transfers/list) \(Incremental\)
+- [Accounts](https://stripe.com/docs/api/accounts/list) \(Incremental\)
 
 ### Data type mapping
 
 The [Stripe API](https://stripe.com/docs/api) uses the same [JSONSchema](https://json-schema.org/understanding-json-schema/reference/index.html) types that Airbyte uses internally \(`string`, `date-time`, `object`, `array`, `boolean`, `integer`, and `number`\), so no type conversions are performed for the Stripe connector.
 
 ### Performance considerations
-
+https://github.com/airbytehq/airbyte/pull/23960
```
Comment (marcosmarxm) [inline docs/integrations/sources/stripe.md:79]: Remove this.

File: docs/integrations/sources/stripe.md [hunk 5]
```diff
@@ -69,19 +69,21 @@ The Stripe source connector supports the following streams:
...
 The Stripe connector should not run into Stripe API limitations under normal usage. [Create an issue](https://github.com/airbytehq/airbyte/issues) if you see any rate limit issues that are not automatically retried successfully.
 
 ## Changelog
 
 | Version | Date       | Pull Request                                             | Subject                                                                                                                                              |
 |:--------|:-----------|:---------------------------------------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------|
+| 3.1.0   | 2023-03-15 | [23960](https://github.com/airbytehq/airbyte/pull/23960) | Add `Accounts` stream
```
Comment (marcosmarxm) [inline docs/integrations/sources/stripe.md:86]: Correct version here too.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Left some comments.

**Comment by haithem-souala** [comment]
@marcosmarxm fixed.

**Comment by haithem-souala** [comment]
> /test connector=connectors/source-stripe
> 
> > 🕑 connectors/source-stripe https://github.com/airbytehq/airbyte/actions/runs/4429641780
> > ❌ connectors/source-stripe https://github.com/airbytehq/airbyte/actions/runs/4429641780
> > 🐛 https://gradle.com/s/vdpzlorqzjhsu
> 
> ### Build Failed
> Test summary info:
> 
> ```
> 	 =========================== short test summary info ============================
> 	 FAILED unit_tests/test_source.py::test_source_streams - assert 25 == 24
> 	 �[31m================== �[31m�[1m1 failed�[0m, �[32m45 passed�[0m, �[33m140 warnings�[0m�[31m in 0.82s�[0m�[31m ==================�[0m
> ```

Fixed!

**Comment by sajarin** [comment]
Hey thanks for the contribution and apologies for the delay! 

I'll be the DRI responsible for reviewing and getting across the finish line. We currently have a backlog of about a dozen PRs, but just wanted to let you know that this PR has been added to the queue and I'll be trying my best to get to this as soon as possible. 

Thanks for the contribution and for being patient :)

**Comment by marcosmarxm** [comment]
Published at https://github.com/airbytehq/airbyte/pull/25136

**Pull request closed without merging**